presto-spi/src/main/java/com/facebook/presto/spi/spiller/TempFileStore.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.spiller;
+
+import com.facebook.presto.common.io.DataSink;
+import io.airlift.slice.SliceInput;
+
+import java.io.IOException;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * An interface for temporary file storage. A temporary file store can be used for the following purpose:
+ * - Spilling
+ * - In Presto-on-Spark, broadcast join can be implemented via writing the build table into a temporary table
+ *   and read back on all executors.
+ *
+ * Note:
+ * - For the spilling use case, the TempFileStore can be either a local filesystem or a disaggregated file system (e.g. S3).
+ * - For the broadcast join use case, the TempFileStore has to be a disaggregated file system; and the temporary file can be read
+ *      from different workers.
+ */
+public interface TempFileStore
+{
+    void initialize();
+
+    TempFile createTemporaryFile()
+            throws IOException;
+
+    SliceInput getSliceInput(String handle)
+            throws IOException;
+
+    void removeTemporaryFile(String handle);
+
+    // Hack!
+    int getFileBufferSize();
```

Comment:
@arhimondr : This is due to `FileSingleStreamSpiller` always reserve the buffer size during the spill time, and only release it upon close. See the long comments: https://github.com/prestodb/presto/blob/029ae8d9962fffcbc793e46c10ed6481d27a19b0/presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpiller.java#L93-L103